lib/Dialect/Arc/Transforms/StripSV.cpp

```diff
@@ -0,0 +1,178 @@
+//===- StripSV.cpp --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/SV/SVOps.h"
+#include "circt/Dialect/Seq/SeqOps.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "llvm/Support/Debug.h"
+#include <variant>
+
+#define DEBUG_TYPE "arc-strip-sv"
+
+using namespace circt;
+using namespace arc;
+
+namespace {
+struct StripSVPass : public StripSVBase<StripSVPass> {
+  void runOnOperation() override;
+  SmallVector<Operation *> opsToDelete;
+  SmallPtrSet<StringAttr, 4> clockGateModuleNames;
+  SmallPtrSet<StringAttr, 4> externModuleNames;
+};
+} // namespace
+
+void StripSVPass::runOnOperation() {
+  auto mlirModule = getOperation();
+  opsToDelete.clear();
+  clockGateModuleNames.clear();
+  externModuleNames.clear();
+
+  auto expectedClockGateInputs =
+      ArrayAttr::get(&getContext(), {StringAttr::get(&getContext(), "in"),
+                                     StringAttr::get(&getContext(), "test_en"),
+                                     StringAttr::get(&getContext(), "en")});
+  auto expectedClockGateOutputs =
+      ArrayAttr::get(&getContext(), {StringAttr::get(&getContext(), "out")});
+  auto i1Type = IntegerType::get(&getContext(), 1);
+
+  for (auto extModOp : mlirModule.getOps<hw::HWModuleExternOp>()) {
+    if (extModOp.getVerilogModuleName() == "EICG_wrapper") {
```

Comment:
This is a pretty ugly artifact of chisel.  It would be nice to do something about it in the design (say have intrinsics it can be implemented with).

Reply:
Yeah I'd like to turn this into an intrinsic on the FIRRTL side, and then maybe have a `seq.clock_gate` operation to capture it at the HW level (or alternatively just as an HW intrinsic).